pkg/scheduler/extender/extender.go

```diff
@@ -329,6 +363,81 @@ func (e *Extender) filter(ctx context.Context, nodes []coreV1.Node, volumes []*g
 	return matchedNodes, failedNodesMap, err
 }
 
+func (e *Extender) score(nodes []coreV1.Node) ([]schedulerapi.HostPriority, error) {
+	ll := e.logger.WithFields(logrus.Fields{
+		"method": "score",
+	})
+
+	var volumeList = &volcrd.VolumeList{}
+	if err := e.k8sClient.ReadList(context.Background(), volumeList); err != nil {
+		err = fmt.Errorf("unable to read AvailableCapacity list: %v", err)
+		return nil, err
+	}
+	ll.Debugf("Got %d volumes", len(volumeList.Items))
+
+	nodeMapping := nodeVolumeCountMapping(volumeList)
+	sortedNoded := getSortedNodes(nodeMapping)
+	ll.Debugf("sorted volumes %v", sortedNoded)
+
+	priorityFromVolumes := nodePrioritize(sortedNoded)
+
+	ll.Debugf("nodes were ranked by their volumes %+v", priorityFromVolumes)
+	hostPriority := make([]schedulerapi.HostPriority, 0, len(nodes))
+	for _, node := range nodes {
+		// set the highest priority if node doesn't have any volumes
+		rank := math.MaxInt16
+		if r, ok := priorityFromVolumes[string(node.GetUID())]; ok {
+			rank = r
+		}
+		hostPriority = append(hostPriority, schedulerapi.HostPriority{
+			Host:  node.GetName(),
+			Score: rank,
+		})
+	}
+	return hostPriority, nil
+}
+
+func nodeVolumeCountMapping(vollist *volcrd.VolumeList) map[string][]volcrd.Volume {
```

Comment:
Shouldn't you take into account only BM CSI volumes?

Reply:
It's BM CSI volumes.